### ISSUE TITLE
Some applications see no internet

### ISSUE THREAD

**Issue by nturns** [Genymobile/gnirehtet#567]
This may be related to #522 . Browser and Aurorastore have internet . Googleplay and VOIP app say there is no internet connection.

**Comment by searinminecraft** [comment]
to be able to think that the device is connected you have to use bluetooth tethering (the remote device doesnt have to provide internet)

**Comment by nturns** [comment]
[searinminecraft](https://github.com/searinminecraft) that doesn't make sense.

**Comment by nturns** [comment]
That doesn't make sense.

**Comment by hugo-akaora** [comment]
> to be able to think that the device is connected you have to use bluetooth tethering (the remote device doesnt have to provide internet)

Hello @searinminecraft can you please elaborate? Where does it need to be enabled? gnirehtet?

**Comment by searinminecraft** [comment]
> > to be able to think that the device is connected you have to use bluetooth tethering (the remote device doesnt have to provide internet)
> 
> Hello @searinminecraft can you please elaborate? Where does it need to be enabled? gnirehtet?

No, you enable Bluetooth tethering on your other Android (typically on network/hotspot settings) or on a Windows device (on the Mobile Hotspot settings, select Bluetooth on the "Share via" dropdown), then the device connected via gnirehtet connects to that Bluetooth device (make sure that you turn on "Internet Access" for that device, but it should automatically detect its providing "internet")

You can also enable data to have the same effect, but theres a possibility that would be prioritized than the VPN connection that gnirehtet uses

The reason why this works is because most if not all Android apps consider a Bluetooth tethering connection as a valid network connection, and some apps do not consider just a VPN connection alone (the connection method gnirehtet use) a valid network connection.

**Comment by Mo-Dabao** [comment]
Same problem here.

My device is Xiaomi15/Android15

When using gnirehtet, some APPs like Wechat, bilibili and even some bank apps work fine, but APPs like Xiaomi APP store, Tonghuashun (Chinese stock app) and some games can't work complaining no internet

**Comment by hugo-akaora** [comment]
Hello, I can confirm that the solution provided by @searinminecraft works well, I use it everyday. Signal doesn't complain anymore that no Internet connection is available. It's not ideal, as it necessitate another device, but it's the only solution I can see. I can add that the Android device that acts as Bluetooth tethering doesn't seem to need to have data connection nor Wi-Fi enabled.

**Comment by nturns** [comment]
To summarize are you saying Bluetooth is just a dummy connection but once Bluetooth is connected the Android apps go ahead and use the gnirehtet VPN for data?

I will admit I previously gave up on gnirehtet and plunked down $10 for Tetrd. It is not perfect on privacy so I'm still following progress on gnirehtet.

**Comment by hugo-akaora** [comment]
@nturns yep, otherwise can't know how Signal is getting my messages, as I don't have any data connection on the bluetooth tethering phone ;-)

**Comment by fisherchase** [comment]
相同问题, 只有浏览器和时间等少量app可以正常使用网络; 而且浏览器下载时提示无网络连接, 插眼蹲个解决方法

**Comment by searinminecraft** [comment]
> 相同问题, 只有浏览器和时间等少量app可以正常使用网络; 而且浏览器下载时提示无网络连接, 插眼蹲个解决方法

You can try [this](#issuecomment-2549647626) workaround and see if it works

**Comment by n4htx** [comment]
On the Q3, it seems to rely solely on being connected to a WiFi network, so when using Gnirehtet, it receives internet perfectly fine, but programs like the main store feed on boot and web browser sometimes ask you to connect to the internet, but other apps like discord, 3rd party browsers, aurora store and others work fine.
